src/server/game/Instances/InstanceScript.cpp

```diff
@@ -181,6 +181,7 @@ void InstanceScript::UpdateDoorState(GameObject* door)
     if (range.first == range.second)
         return;
 
+    // more than one condition can be needed for the door to open
     bool open = true;
     for (; range.first != range.second && open; ++range.first)
```

Comment:
if the check were to be removed then the PR would work as advertised - multiple conditions for the same door would turn into an AND instead of only abiding the last condition.

I see no issue with making that change.